cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -80,7 +83,8 @@ protected final void startScheduler() {
    */
   protected final void stopScheduler() {
     try {
-      delegate.stop();
+      timeScheduler.stop();
```

Comment:
you need to catch exceptions from each stop(). Otherwise, if the first stop() throws an exception, the second one will not be called.